File: app/src/main/java/de/christinecoenen/code/zapp/model/ChannelModel.java
package de.christinecoenen.code.zapp.model;

import java.io.Serializable;

public class ChannelModel implements Serializable {

	private String id;
	private String name;
	private String subtitle;
	private String streamUrl;
	private int drawableId;

	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getSubtitle() {
		return subtitle;
	}

	public void setSubtitle(String subtitle) {
		this.subtitle = subtitle;
	}

	public String getStreamUrl() {
		return streamUrl;
	}

	public void setStreamUrl(String streamUrl) {
		this.streamUrl = streamUrl;
	}

	public int getDrawableId() {
		return drawableId;
	}

	public void setDrawableId(int drawableId) {
		this.drawableId = drawableId;
	}

	@Override
	public String toString() {
		return "ChannelModel{" +
				"id='" + id + '\'' +
				", name='" + name + '\'' +
				", subtitle='" + subtitle + '\'' +
				", streamUrl='" + streamUrl + '\'' +
				", drawableId=" + drawableId +
				'}';
	}
}

File: app/src/main/java/de/christinecoenen/code/zapp/model/ChannelModel.kt
package de.christinecoenen.code.zapp.model

import android.content.Intent
import android.graphics.Color
import android.net.Uri

import java.io.Serializable

data class ChannelModel(var id: String = "",
						var name: String = "",
						var subtitle: String? = null,
						var streamUrl: String = "",
						var drawableId: Int = 0,
						var color: Int = Color.BLACK) : Serializable {

	val videoShareIntent: Intent
		get() {
			val videoIntent = Intent(Intent.ACTION_VIEW)
			videoIntent.setDataAndType(Uri.parse(streamUrl), "video/*")
			return videoIntent
		}
}
